server/src/main/java/org/spine3/server/entity/storage/StorageFields.java

```diff
@@ -0,0 +1,203 @@
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.entity.storage;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.LinkedListMultimap;
+import com.google.common.collect.Multimap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.spine3.annotations.Internal;
+import org.spine3.server.entity.Entity;
+
+import java.beans.BeanInfo;
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.Multimaps.synchronizedListMultimap;
+import static java.lang.String.format;
+
+/**
+ * A utility for generating the Storage Fields {@linkplain Map}.
+ *
+ * <p>All the methods of the passed {@link Entity} that fit
+ * <a href="http://download.oracle.com/otndocs/jcp/7224-javabeans-1.01-fr-spec-oth-JSpec/">the Java Bean</a>
+ *  getter spec are considered Storage Fields.
+ *
+ * <p>When passing an instance of an already known {@link Entity} type, the getters are retrieved
+ * from a cache and are not updated.
+ *
+ * <p>There are several excluded methods, which are never taken into account and are
+ * <b>not</b> considered Storage Fields:
+ * <ul>
+ *     <li>{@link Object#getClass()}
+ *     <li>{@link Entity#getId()}
+ *     <li>{@link Entity#getState()}
+ *     <li>{@link Entity#getDefaultState()}
+ *     <li>{@link org.spine3.server.entity.EntityWithLifecycle#getLifecycleFlags() EntityWithLifecycle#getLifecycleFlags()}
+ *     <li>{@link org.spine3.server.aggregate.Aggregate#getBuilder() Aggregate#getBuilder()}
+ * </ul>
+ *
+ * <p>Note: if creating a getter method with a name which intersects with one of these method
+ * names, your getter method will also <b>not</b> be considered a Storage Field.
+ *
+ * @author Dmytro Dashenkov
+ * @see Column
+ */
+public class StorageFields {
+
+    private static final String NON_PUBLIC_CLASS_WARNING =
+            "Passed entity class %s is not public. Storage fields won't be extracted.";
+
+    /**
+     * An immutable {@link java.util.Set set} of excluded methods which fit the getter pattern but
+     * are not Storage Fields.
+     */
+    private static final ImmutableSet<String> EXCLUDED_METHODS =
```

Comment:
Instead, I'd have a nested enum (with a descent description for each of the items) and an API telling whether some method is excluded.